js/src/Figure.js

```diff
@@ -635,6 +635,12 @@ var Figure = widgets.DOMWidgetView.extend({
         this.title.text(this.model.get("title"));
     },
 
+    remove: function() {
+        Figure.__super__.remove.apply(this, arguments);
+        this.mark_views.remove()
+        this.axis_views.remove()
```

Comment:
Should they be called in the reverse order? Generally for a `destructor`, we `destory` the `child` first and then the `parent`, no?